src/Corax/IndexSearcher/MultiTermMatches/IndexSearcher.MultiTermMatches.Common.cs

```diff
@@ -37,6 +38,7 @@ private MultiTermMatch MultiTermMatchBuilder<TTermProvider>(in FieldMetadata fie
         {
             seekKey = _fieldsTree.Llt.AcquireCompactKey();
             seekKey.Set(seekTerm.AsReadOnlySpan());
+            seekKey.ChangeDictionary(terms.DictionaryId);
         }
```

Comment:
Why do we need to change the dictionary or better phrased, if that is needed, why it didn't fail before?

Reply:
This is so the `Inspect` call will tell us what is the actual key that is being used in terms.
That shouldn't cause any affect unless you are using query timings.